### PR TITLE
MULTIARCH-1411,1393: Updates IBM Z for OCP 4.9 

### PR DESCRIPTION
**Pull request by SNiemann15**
- OCP version for cherry-picking: enterprise-4.9

- Jira:
   https://issues.redhat.com/browse/MULTIARCH-1411
   https://issues.redhat.com/browse/MULTIARCH-1393

- Bugzilla for Multi-NIC update for all platforms: https://bugzilla.redhat.com/show_bug.cgi?id=2000583
 
- Google doc: https://docs.google.com/document/d/1pqZz7o5BBIAUVMPj4fVna_h26BfPBmIHlGo116fABtI/edit#

- Preview
   - [Installing a cluster with z/VM on IBM Z and LinuxONE](https://deploy-preview-36250--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z.html)
   - [Installing a cluster with z/VM on IBM Z and LinuxONE in a restricted network](https://deploy-preview-36250--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z.html)
   - [Installing a cluster with RHEL KVM on IBM Z and LinuxONE](https://deploy-preview-36250--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-ibm-z-kvm.html)
   - [Installing a cluster with RHEL KVM on IBM Z and LinuxONE in a restricted network](https://deploy-preview-36250--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_ibm_z/installing-restricted-networks-ibm-z-kvm.html)

- QE review IBM Z: Holger Wolf, Wolfgang Voesch
- QE review for Multi-NIC: Muhammad Adeel (IBM Z) Micah Abbott (x86)

### CODE REVIEW

**Review by madeelrh — COMMENTED** [review]
If you specify multiple ip kernel arguments, 
instead of IP in caps.

**Review by madeelrh — COMMENTED** [review]
If the additional network interface gateway is different from the primary network interface, the default gateway must be the primary network interface.

here interface can be removed:
If the additional network gateway is different from the primary network gateway, the default gateway must be the primary network gateway.

**Comment by madeelrh** [comment]
lgtm

File: modules/installation-user-infra-machines-static-network.adoc [hunk 1]
```diff
@@ -56,12 +73,25 @@ nameserver=4.4.4.41
 ----
 
 a|Specify multiple network interfaces by specifying multiple `ip=` entries.
+
+If the additional network gateway is different from the primary network gateway, the default gateway must be the primary network gateway.
```
Comment (SNiemann15) [inline modules/installation-user-infra-machines-static-network.adoc:77]: @miabbott Could you please take a look and confirm that it's correct for x86. Thanks

**Comment by wvoesch** [comment]
lgtm

**Comment by alkleinrh** [comment]
lgtm

**Comment by ktania46** [comment]
The updates are valid for IBM Power too

OCP version for cherry-picking: enterprise-4.9

Jira:
https://issues.redhat.com/browse/MULTIARCH-1410

Bugzilla for Multi-NIC update for all platforms: https://bugzilla.redhat.com/show_bug.cgi?id=2000583

Google doc: https://docs.google.com/document/d/1ltvH5VztEmgAuFoIaWrbBR1i8rvmr55KwPwUF7AQXqY/edit?disco=AAAAPpGoPEY

Preview

QE review: Pravin D-Silva , Mick Tarsel

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -387,6 +387,10 @@ endif::bare[]
 The IP address blocks for machines.
 
 If you specify multiple IP address blocks, the blocks must not overlap.
+
+ifdef::ibm-z[]
+If you specify multiple ip kernel arguments, the `machineNetwork.cidr` must be the CIDR of the primary network.
+endif::ibm-z[]
 |An array of objects. For example:
```
Comment (ktania46) [inline modules/installation-configuration-parameters.adoc:394]: ifdef::ibm-power[]
If you specify multiple ip kernel arguments, the `machineNetwork.cidr` must be the CIDR of the primary network.
endif::ibm-power[]

**Comment by mtarsel** [comment]
lgtm

File: modules/installation-user-infra-machines-static-network.adoc [hunk 3]
```diff
@@ -56,12 +82,25 @@ nameserver=4.4.4.41
...
+
+----
+ip=::10.10.10.254::::
+----
+Configure the route for the additional network:
+
+----
+rd.route=20.20.20.0/24:20.20.20.254:enp2s0
+---- 
+
```
Comment (miabbott) [inline modules/installation-user-infra-machines-static-network.adoc:103]: Can we break this section into it's own table row? I don't want to mix up the example of specifying multiple `ip=` entries with providing an additional routing entry.

**Review by miabbott — APPROVED** [review]
Thanks for working with my suggested changes!

LGTM!

File: modules/installation-bare-metal-config-yaml.adoc [hunk 4]
```diff
@@ -227,6 +228,19 @@ Simultaneous multithreading (SMT) is enabled by default. If SMT is not enabled i
...
+ifdef::ibm-z,ibm-z-kvm[]
++
+[NOTE]
+====
+Simultaneous multithreading (SMT) is enabled by default. If SMT is not available on your {product-title} nodes, the `hyperthreading` parameter has no effect.
+====
++
+[IMPORTANT]
+====
+If you disable `hyperthreading`, whether on your {product-title} nodes or in the `install-config.yaml`, ensure that your capacity planning accounts for the dramatically decreased machine performance.
```
Comment (kalexand-rh) [inline modules/installation-bare-metal-config-yaml.adoc:241]:
```suggestion
If you disable `hyperthreading`, whether on your {product-title} nodes or in the `install-config.yaml` file, ensure that your capacity planning accounts for the dramatically decreased machine performance.
```

> Follow the file name, file name extension, or file type with the word file,

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -387,6 +387,10 @@ endif::bare[]
 The IP address blocks for machines.
 
 If you specify multiple IP address blocks, the blocks must not overlap.
+
+ifdef::ibm-z,ibm-power[]
+If you specify multiple ip kernel arguments, the `machineNetwork.cidr` must be the CIDR of the primary network.
```
Comment (kalexand-rh) [inline modules/installation-configuration-parameters.adoc:392]:
```suggestion
If you specify multiple IP kernel arguments, the `machineNetwork.cidr` value must be the CIDR of the primary network.
```

> Generally, specify the keyword type after the keyword.

File: modules/installation-user-infra-machines-static-network.adoc [hunk 6]
```diff
@@ -56,12 +82,29 @@ nameserver=4.4.4.41
...
 
 a|Specify multiple network interfaces by specifying multiple `ip=` entries.
+
 a|
 ----
 ip=10.10.10.2::10.10.10.254:255.255.255.0:core0.example.com:enp1s0:none
 ip=10.10.10.3::10.10.10.254:255.255.255.0:core0.example.com:enp2s0:none
 ----
 
+a|Optional: You can configure routes to additional networks using `rd.route=`.
```
Comment (kalexand-rh) [inline modules/installation-user-infra-machines-static-network.adoc:92]: Maybe
```suggestion
a|Optional: You can configure routes to additional networks by setting an `rd.route=` value.  
```

**Review by kalexand-rh — COMMENTED** [review]
I just see a few small style tweaks, but this is looking good!

File: modules/installation-bare-metal-config-yaml.adoc (hunk 4 above)
Comment (SNiemann15) [reply modules/installation-bare-metal-config-yaml.adoc:241]: I'm also adding "file" in line 229 to be consistent

**Pull request merged**